src/web/app/views/beambox/Photo-Edit-Panel.tsx

```diff
@@ -10,643 +9,549 @@ import Modal from 'app/widgets/Modal';
-                break;
-            case 'invert':
-                this._handleInvertAndComplete();
-                break;
-            case 'stamp':
-                this._handleStamp();
-                break;
-            default:
-                break;
-        }
-    }
+const opencvWS = new OpenCVWebSocket();
+let LANG = i18n.lang.beambox.photo_edit_panel;
+const updateLang = () => {
+  LANG = i18n.lang.beambox.photo_edit_panel;
+};
 
-    async _handlePreprocess() {
-        let imgBlobUrl = this.state.origSrc;
-        try {
-            const res = await fetch(imgBlobUrl)
-            const blob = await res.blob();
-            const arrayBuffer = await new Response(blob).arrayBuffer(); 
-            const image = await jimp.read(arrayBuffer);
-            let w = image.bitmap.width;
-            let h = image.bitmap.height;
-            if (['sharpen', 'curve'].includes(this.props.mode)){
-                if( Math.max(image.bitmap.width, image.bitmap.height) > 600 ) {
-                    console.log('Down Sampling');
-                    if (image.bitmap.width >= image.bitmap.height) {
-                        image.resize(600, jimp.AUTO);
-                    } else {
-                        image.resize(jimp.AUTO, 600);
-                    }
-                    const buffer = await image.getBufferAsync(jimp.MIME_PNG);
-                    const newBlob = new Blob([buffer]);
-                    const src = URL.createObjectURL(newBlob);
-                    if (this.state.src !== this.state.origSrc) {
-                        URL.revokeObjectURL(this.state.src);
-                    }
-                    this.state.previewSrc= src;
-                    this.setState({
-                        imageWidth: w,
-                        imageHeight: h,
-                        previewSrc: src,
-                        src: src,
-                        hadPreprocessDone: true
-                    });
-                } else {
-                    this.setState({
-                        imageWidth: w,
-                        imageHeight: h,
-                        previewSrc: imgBlobUrl,
-                        src: imgBlobUrl,
-                        hadPreprocessDone: true
-                    });
-                }
-            } else if (this.props.mode === 'crop') {
-                Progress.popById('photo-edit-processing');
-
-                this.setState({
-                    imageWidth: w,
-                    origWidth: w,
-                    imageHeight: h,
-                    origHeight: h,
-                    hadPreprocessDone: true
-                });
-            } else {
-                throw new Error(`Unknown Mode ${this.props.mode}`); 
-
-            }
-            
-        } catch (err) {
-            console.log(err)
-            Progress.popById('photo-edit-processing');
-        }
+interface IProps {
+  element: HTMLElement,
+  src: string,
+  mode: 'sharpen' | 'crop' | 'curve',
```

Comment:
duplicate definition, could we define it somewhere making it reused?
https://www.typescriptlang.org/docs/handbook/literal-types.html#string-literal-types

Reply:
I will add a literal type and export it.